### PR TITLE
feat(Scripts/Commands): deserter add/remove bg/instance allow names

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10826

## Tests Performed:

- Not-Tested 
- 

## How to Test the Changes:

1. .deserter bg add 3600 playername
2. .deserter instance add 3600 playername
3. .deserter bg remove all
4. .deserter instance remove all

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Lynethris** [comment]
Tested and work:

- [X] .deserter bg add 3600 playername
- [X] .deserter instannce add 3600 playername
- [X] .deserter bg remove all
- [X] .deserter instance remove all

**Comment by Nyeriah** [comment]
Hey,

I am sorry but this PR doesn't seem to address the issue described in that ticket.

The point of allowing names is to
1) Allow using it on offline targets (not covered in this PR, they still need to be online)
2) Allow usuing it from console

Right now, as it is, there's not much of a benefit with the name addition, as teleporting to players to target them is not the issue, but debuffing them while they're offline is.

**Comment by Lynethris** [comment]
I'm sorry. I didn't look the initial issue closed by the PR. yesterday, Temper had asked me to test the console aspect, but we misunderstood each other and concluded that it is not intentionally covered by this PR.

I have redone some tests this morning, and I can also confirm that the commands doesn't work if the player is offline (even with remove all).

**Comment by IntelligentQuantum** [comment]
@Winfidonarleyan Do I have to get the offline player from the database with the new system?

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 1]
```diff
@@ -90,24 +92,16 @@ class deserter_commandscript : public CommandScript
     * .deserter bg add 3600 (one hour)
     * @endcode
     */
-    static bool HandleDeserterAdd(ChatHandler* handler, uint32 time, bool isInstance)
+    static bool HandleDeserterAdd(ChatHandler* handler, uint32 time, Player* targetPlayer, bool isInstance)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_deserter.cpp:95]:
```suggestion
    static bool HandleDeserterAdd(ChatHandler* handler, uint32 time, Optional<PlayerIdentifier> target, bool isInstance)
```

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 2]
```diff
@@ -139,43 +135,108 @@ class deserter_commandscript : public CommandScript
     * .deserter bg remove
     * @endcode
     */
-    static bool HandleDeserterRemove(ChatHandler* handler, bool isInstance)
+    static bool HandleDeserterRemove(ChatHandler* handler, Player* targetPlayer, bool isInstance)
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_deserter.cpp:138]:
```suggestion
    static bool HandleDeserterRemove(ChatHandler* handler, Optional<PlayerIdentifier> target, bool isInstance)
```

**Comment by Winfidonarleyan** [comment]
For this need use `Optional<PlayerIdentifier>`

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan Do I have to get the offline player from the database with the new system?

Yes, need use `Optional<PlayerIdentifier>`

**Comment by IntelligentQuantum** [comment]
@Winfidonarleyan Hmm with Optional<PlayerIdentifier> target i dont have access to AddAura

**Pull request closed without merging**